cmd/applications-rp/cmd/root.go

```diff
@@ -0,0 +1,139 @@
+	etcdclient "go.etcd.io/etcd/client/v3"
+	runtimelog "sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/radius-project/radius/pkg/armrpc/builder"
+	"github.com/radius-project/radius/pkg/armrpc/hostoptions"
+	metricsservice "github.com/radius-project/radius/pkg/metrics/service"
+	profilerservice "github.com/radius-project/radius/pkg/profiler/service"
+	"github.com/radius-project/radius/pkg/recipes/controllerconfig"
+	"github.com/radius-project/radius/pkg/server"
+	"github.com/radius-project/radius/pkg/trace"
+
+	"github.com/radius-project/radius/pkg/ucp/data"
+	"github.com/radius-project/radius/pkg/ucp/dataprovider"
+	"github.com/radius-project/radius/pkg/ucp/hosting"
+	"github.com/radius-project/radius/pkg/ucp/ucplog"
+
+	corerp_setup "github.com/radius-project/radius/pkg/corerp/setup"
+	daprrp_setup "github.com/radius-project/radius/pkg/daprrp/setup"
+	dsrp_setup "github.com/radius-project/radius/pkg/datastoresrp/setup"
+	msgrp_setup "github.com/radius-project/radius/pkg/messagingrp/setup"
+)
+
+const serviceName = "radius"
+
+var rootCmd = &cobra.Command{
+	Use:   "applications-rp",
+	Short: "Applications.* Resource Provider Server",
+	Long:  `Server process for the Applications.* Resource Provider (applications-rp).`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		configFilePath := cmd.Flag("config-file").Value.String()
+		options, err := hostoptions.NewHostOptionsFromEnvironment(configFilePath)
+		if err != nil {
+			return err
+		}
+
+		hostingSvc := []hosting.Service{}
+
+		metricOptions := metricsservice.NewHostOptionsFromEnvironment(*options.Config)
+		metricOptions.Config.ServiceName = serviceName
+		if metricOptions.Config.Prometheus.Enabled {
+			hostingSvc = append(hostingSvc, metricsservice.NewService(metricOptions))
+		}
+
+		profilerOptions := profilerservice.NewHostOptionsFromEnvironment(*options.Config)
+		if profilerOptions.Config.Enabled {
+			hostingSvc = append(hostingSvc, profilerservice.NewService(profilerOptions))
+		}
+
+		logger, flush, err := ucplog.NewLogger(serviceName, &options.Config.Logging)
+		if err != nil {
+			return err
+		}
+		defer flush()
+
+		// Must set the logger before using controller-runtime.
+		runtimelog.SetLogger(logger)
+
+		if options.Config.StorageProvider.Provider == dataprovider.TypeETCD &&
+			options.Config.StorageProvider.ETCD.InMemory {
+			// For in-memory etcd we need to register another service to manage its lifecycle.
+			//
+			// The client will be initialized asynchronously.
+			logger.Info("Enabled in-memory etcd")
+			client := hosting.NewAsyncValue[etcdclient.Client]()
+			options.Config.StorageProvider.ETCD.Client = client
+			options.Config.SecretProvider.ETCD.Client = client
+
+			hostingSvc = append(hostingSvc, data.NewEmbeddedETCDService(data.EmbeddedETCDServiceOptions{ClientConfigSink: client}))
+		}
+
+		builders, err := builders(options)
+		if err != nil {
+			log.Fatal(err) //nolint:forbidigo // this is OK inside the main function.
```

Comment:
Will remove this. We don't have a reason to use `log.Fatal` or similar constructs in this code.

Reply:
I will do this now :). Sometimes I comment on my own PRs.